common/iputils.hh

```diff
@@ -722,13 +722,31 @@ public:
     return ret;
   }
 
-  //<! Removes key from TreeMap. This does not clean up the tree.
-  void erase(const key_type& key) {
+  void cleanup_tree(TreeNode* node)
+  {
+    // only cleaup this node if it has no children and node4 and node6 are both empty
+    if (!(node->left || node->right || node->node6 || node->node4)) {
+      // get parent node ptr
+      TreeNode* parent = node->parent;
+      // delete this node
+      if (parent) {
+	if (parent->left.get() == node)
+	  parent->left.reset();
+	else
+	  parent->right.reset();
+	// now recurse up to the parent
+	cleanup_tree(parent);
+      }
+    }
+  }
+  
+  //<! Removes key from TreeMap. This optionally cleans up the tree.
```

Comment:
the cleanup here is slightly misleading, and I think it should probably be moved to ctor as parameter instead, since it does not *really* cleanup the whole tree. either the tree cleans itself or it doesn't. inconsistently using this can lead into rather funny tree.